src/workflow.ts

```diff
@@ -700,10 +701,19 @@ export class WorkflowContextImpl extends DBOSContextImpl implements WorkflowCont
           }
           this.resultBuffer.set(funcId, readOutput);
         } else {
-          // Synchronously record the output of write transactions and obtain the transaction ID.
-          const pg_txn_id = await this.recordOutputTx<R>(client, funcId, txn_snapshot, result);
-          tCtxt.span.setAttribute("pg_txn_id", pg_txn_id);
-          this.resultBuffer.clear();
+          try {
+            // Synchronously record the output of write transactions and obtain the transaction ID.
+            const pg_txn_id = await this.recordOutputTx<R>(client, funcId, txn_snapshot, result);
+            tCtxt.span.setAttribute("pg_txn_id", pg_txn_id);
+            this.resultBuffer.clear();
+          } catch (error) {
+            if (this.#dbosExec.userDatabase.isFailedSqlTransactionError(error)) {
+              this.logger.error(`Postgres aborted the ${txn.name} @Transaction of Workflow ${workflowUUID}, but the function did not raise an exception. Please ensure that if a transaction aborts from an exception, that exception is allowed to escape the @Transaction method.`);
```

Comment:
I think we want to emphasize that if a transaction aborts from an exception, we **must** raise an exception.

```suggestion
              this.logger.error(`Postgres aborted the ${txn.name} @Transaction of Workflow ${workflowUUID}, but the function did not raise an exception. Please ensure that if a transaction aborts from an exception, it must raise an exception.`);
```